parsers/ES.py

```diff
@@ -163,40 +234,41 @@ def fetch_production(
 ) -> list[dict]:
     check_valid_parameters(zone_key, session, target_datetime)
     ses = session or Session()
+    data_mapping = PRODUCTION_PARSE_MAPPING.copy()
+
+    ## Production mapping override for Canary Islands
+    # NOTE the LNG terminals are not built yet, so power generated by "gas" or "cc" in ES-CN domain is actually using oil.
+    # Recheck this every 6 months and move to gas key if there has been a change.
+    # Last checked: 2022-06-27
+    if zone_key.split("-")[1] == "CN":
+        data_mapping["gas"] = "oil"
+        data_mapping["cc"] = "oil"
+
+    if zone_key == "ES-IB-ME" or zone_key == "ES-IB-FO":
+        data_mapping["gas"] = "oil"
+    if zone_key == "ES-IB-IZ":
+        data_mapping["die"] = "gas"
+
     island_data = fetch_island_data(zone_key, ses, target_datetime)
     productionEventList = ProductionBreakdownList(logger)
-
     for event in island_data:
-        production = ProductionMix()
+
         storage = StorageMix()
+        if "hid" in event:
+            storage.add_value("hydro", -event["hid"])
 
-        for mode in PRODUCTION_MAPPING.keys():
-            if mode in event.__dir__():
-                value = getattr(event, mode)
-                ## Production mapping override for Canary Islands
-                # NOTE the LNG terminals are not built yet, so power generated by "gas" or "combined" in ES-CN domain is actually using oil.
-                # Recheck this every 6 months and move to gas key if there has been a change.
-                # Last checked: 2022-06-27
-                fuel_mapping = PRODUCTION_MAPPING
-                if zone_key.split("-")[1] == "CN":
-                    fuel_mapping = {
-                        **PRODUCTION_MAPPING,
-                        "gas": "oil",
-                        "combined": "oil",
-                    }
-                    # Reset gas to 0, as it is now oil, normally this should be skipped or set to None but to conform to how REE does it, we set it to 0
-                    # We should aim to change this upstream in the future to make it uniform with other parsers going forward.
-                    production.gas = 0
-                production.add_value(fuel_mapping[mode], value, True)
-                # Zone specific override for El Hierro
-                # Hydro response is hydro storage, not hydro production
-                if zone_key == "ES-CN-HI" and mode == "hydraulic":
-                    storage.hydro = -value
-                    production.hydro = 0
+        timestamp = event.pop("timestamp").astimezone(timezone.utc)
```

Comment:
Why are we poping the timestamp (that is actually a datetime) instead of just assigning it later on?

And as we are adding a timezone in the preprocess step we don't need to do astimezone here.